app/services/case_self_assign_service.rb

```diff
@@ -0,0 +1,29 @@
+class CaseSelfAssignService
+  attr_reader :result, :assignment
+
+  def initialize(kase:, role:, user:)
+    @case = kase
+    @role = role
+    @user = user
+    @result = :incomplete
+  end
+
+  def call
+    Assignment.connection.transaction do
+      target_team = @user.responding_teams.first
+      @assignment = @case.assignments.new(team: target_team, role: @role, user: @user)
+      if @assignment.valid?
+        managing_team = @user.responding_teams.first
+        @case.state_machine.assign_responder! acting_user: @user, acting_team: managing_team, target_team: target_team
+        @assignment.case.state_machine.accept_approver_assignment!(acting_user: @user, acting_team: target_team)
+        @assignment.accepted!
```

Comment:
Out of interest what does this `.accepted!` method do?

Reply:
Part of the Rails built in ActiveRecord support for enum - it enables magic bang and predicate methods to call on the model instance that are derived from the enum names and set the relevant field to the corresponding value
https://api.rubyonrails.org/v5.2.3/classes/ActiveRecord/Enum.html